ReactiveFeedback/Feedback.swift

```diff
@@ -94,8 +94,10 @@ public struct Feedback<State, Event> {
         predicate: @escaping (State) -> Bool,
         effects: @escaping (State) -> Effect
     ) where Effect.Value == Event, Effect.Error == NoError {
-        self.init(deriving: { $0.filter(predicate) },
-                  effects: effects)
+        self.init(deriving: { $0 },
+                  effects: { state -> SignalProducer<Event, NoError> in
+                      predicate(state) ? effects(state).producer : .empty                      
```

Comment:
I guess this new implementation is just the same as `init(lensing: { $0 } effects: effects)`.

By modifying to current code f4905c7 , we lost the original filtering logic that once didn't cancel previous effect when predicate failed.